pandas/core/internals/concat.py

```diff
@@ -77,10 +77,16 @@ def _concatenate_array_managers(
     # reindex all arrays
     mgrs = []
     for mgr, indexers in mgrs_indexers:
+        axis1_needs_copy = True
```

Comment:
nitpick again: elsewhere in this file we do basically the same thing with `made_copy = False`